src/map.cpp

```diff
@@ -3666,9 +3666,12 @@ void map::shoot( const tripoint &p, projectile &proj, const bool hit_items )
     };
 
     ter_id terrain = ter( p );
-    if( terrain == t_wall_wood_broken ||
-        terrain == t_wall_log_broken ||
-        terrain == t_door_b ) {
+    if( terrain->has_flag( "TFLAG_NO_SHOOT" ) ) {
```

Comment:
Since you've added "TFLAG_NO_SHOOT" as a bitflag, use it instead:
```cpp
if( terrain->has_flag( TFLAG_NO_SHOOT ) ) { 
```

Reply:
Thanks!